cli/src/semgrep/git.py

```diff
@@ -68,20 +69,15 @@ def __init__(self, base_commit: str, is_mergebase: bool = False) -> None:
         Raises Exception if
         - cwd is not in git repo
         - base_commit is not valid git hash
-        - there are tracked files with pending changes
-        - there are untracked files that will be overwritten by a file in the base commit
         """
         self._base_commit = base_commit
         self._is_mergebase = is_mergebase
-        self._dirty_paths_by_status: Optional[Dict[str, List[Path]]] = None
 
         try:
             # Check commit hash exists
             git_check_output(["git", "cat-file", "-e", base_commit])
 
             self.status = self._get_git_status()
-            self._abort_on_pending_changes()
-            self._abort_on_conflicting_untracked_paths(self.status)
         except subprocess.CalledProcessError as e:
```

Comment:
The way git UI renders the indentation on line 84 confused me so much as to what's happening

Reply:
These two lines would check for if the git state was clean (no unstaged/staged files, or untracked files that conflict with the baseline commit), as before we were performing a git checkout in the same working tree. Now we just create a new working tree at the baseline, so we don't need to check these.

Reply:
Oh sorry this was just a gripe about github :D